### PR TITLE
Add `Tooltip` to `AccessibleButton`

### PR DESCRIPTION
**Pull request by florianduros**
Signed-off-by: Michael Telatynski <7t3chguy@gmail.com>

## Checklist

-   [x] Tests written for new code (and old code if feasible).
-   [x] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [x] Linter and other CI checks pass.
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md)).

Part of https://github.com/element-hq/element-web/issues/26762
Started in https://github.com/matrix-org/matrix-react-sdk/pull/12143 . New PR and branch because I rollbacked some changes.

This PR:
- Add a tooltip to `AccessibleButton`
   When a title attribute is added on a div/button/etc, the browser displays a built in tooltip when the element is hovered. Instead we display a compound tooltip to be more accessible and to harmonize (visual and behaviour) all the element tooltips.
- Deprecate `AccessibleTooltipButton`, `AccessibleButton` with the new integrated tooltip is enough and use the compoud tooltip. `AccessibleTooltipButton` uses the old react-sdk tooltip.
  In a new PR, the `AccessibleTooltipButton` usage will be migrated to `AccessibleButton`.

### CODE REVIEW

File: src/accessibility/context_menu/ContextMenuButton.tsx [hunk 1]
```diff
@@ -36,7 +36,6 @@ export const ContextMenuButton = forwardRef(function <T extends keyof JSX.Intrin
             {...props}
             onClick={onClick}
             onContextMenu={onContextMenu ?? onClick ?? undefined}
-            title={label}
```
Comment (richvdh) [inline src/accessibility/context_menu/ContextMenuButton.tsx:39]: why does this no  longer need a title?

File: src/components/views/elements/AccessibleButton.tsx [hunk 2]
```diff
@@ -86,6 +87,14 @@ type Props<T extends keyof JSX.IntrinsicElements> = DynamicHtmlElementProps<T> &
      * Event handler for button activation. Should be implemented exactly like a normal `onClick` handler.
      */
     onClick: ((e: ButtonEvent) => void | Promise<void>) | null;
+    /**
+     * The tooltip to show on hover or focus.
+     */
+    title?: string;
+    /**
+     * The caption to show within a tooltip as above, only valid when used in conjunction with `title`.
```
Comment (richvdh) [inline src/components/views/elements/AccessibleButton.tsx:95]: What's a caption, in this sense? What does it mean to "show within a tooltip"?

Also, comma splice. s/,/./

File: test/components/views/beacon/__snapshots__/DialogSidebar-test.tsx.snap [hunk 3]
```diff
@@ -14,11 +14,12 @@ exports[`<DialogSidebar /> renders sidebar correctly with beacons 1`] = `
         View list
       </h4>
       <div
+        aria-label="Close sidebar"
         class="mx_AccessibleButton mx_DialogSidebar_closeButton"
+        data-state="closed"
```
Comment (richvdh) [inline test/components/views/beacon/__snapshots__/DialogSidebar-test.tsx.snap:19]: where are these new `data-state` attributes coming from?

File: src/accessibility/context_menu/ContextMenuButton.tsx (hunk 1 above)
Comment (florianduros) [reply src/accessibility/context_menu/ContextMenuButton.tsx:39]: Previously, despite the title attribute, the browser wasn't displayed in this case. (For example the user menu (avatar) at the top left of EW). I removed the title attribute to keep the same behaviour with the new `AccessibleButton` change. In term of accessibility, the `aria-label` attribute is already provided.

File: src/components/views/elements/AccessibleButton.tsx (hunk 2 above)
Comment (florianduros) [reply src/components/views/elements/AccessibleButton.tsx]: I added more informations

**Review by richvdh — APPROVED** [review]
LGTM

**Pull request merged**